cloudsync/tasks.py

```diff
@@ -209,3 +295,21 @@ def parse_content_metadata(response):
         content_length = int(content_length)
 
     return file_name, content_type, content_length
+
+
+def extract_title(filename):
+    """
+    Parses the filename for required course information
+    Args:
+        filename(str): The name of the video file, in format
+        'MIT-<course#>-<year>-<semester>-lec-mit-0000-<recording_date>-<time>-<session>.mp4'
+
+    Returns:
+        A string which should match a collection title: MIT-<course>-<session>-<year>-<semester>
+    """
+    rx = r'([A-Z]+)-([\w\.\s]+)-((\d{4})-(Fall|Spring)-)?lec-mit-0000-(\w+)-(\d+)(-([L\d\-]+))?(_\d+)?.*\.\w'
```

Comment:
I think we should keep in mind what Elaine's inputs look like on the front end. I know that she is using things like course number and section number, but those are by convention and aren't enforced to be in the formats you are looking for. 

I think it would be safer, easier and more flexible to just extract the two strings that Elaine inputs as parameters to the 1Beyond script, something like:

collection_prefix = "^(.*)lec-mit-0000-"
collection_suffix = "lec-mit-0000-(\w+)-(\d+)(-([L\d\-]+))" #do we really need all these parens?
collection_title = collection_prefix + collection_suffix